servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/SubscribablePublisher.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright © 2019 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.concurrent.api;
+
+import io.servicetalk.concurrent.PublisherSource;
+
+/**
+ * A {@link Publisher} that is also a {@link PublisherSource} and hence can be subscribed.
+ * <p>
+ * Typically, this will be used to implement a {@link Publisher} that does not require an additional allocation when
+ * converting to a {@link PublisherSource} via {@link SourceAdapters#toSource(Publisher)}.
+ *
+ * @param <T> Type of the items emitted by this {@code SubscribablePublisher}.
+ */
+public abstract class SubscribablePublisher<T> extends Publisher<T> implements PublisherSource<T> {
```

Comment:
Can we avoid making this part of the public API? The only use case seems to be in `SpliceFlatStreamToPackedSingle` and it seems like we can just return an anonymous `Publisher<T>` and remove this altogether.

Reply:
yes, I missed this when I moved it over, we should definitely not have this as part of the public API

Reply:
and yes it can be inlined